### PR TITLE
parse primitives and bugfix on SFW query

### PR DESCRIPTION
**Pull request by am357**
*Issue #, if available:* #108 #137 

*Description of changes:*

This includes the following fixes that reflect in successful execution of corresponding conformance tests:
- Fix parsing `VALUES` with one item rows
- Include '\\' to string liternal
- Parse `DATE`, `TIME`, and `TIMESTAMP`
- Parse correlated JOIN
- Parse outer join as full join
- Make `FROM` required in SFW query

In addition it includes addition of SQL Parameter token `?` to the Lexer; this is a prerequisite for preprocessing of  SQL parameters in queries for resolving their index in AST.

Note: number of conformance test failures shows an increase in the conformance test report; there are two reasons for this:

1. New tests related to `DATA`, `TIME`, and `TIMESTAMP` have been introduced, and the tests related to receiving a parameter for data and time are failing. We've decided to consider these tests as `pass` in for `Syntax` test which is a new category introduced in [partiql-tests/#18](https://github.com/partiql/partiql-tests/pull/18) which hasn't been merged yet.
2. There are 3 `IS` operator tests that are failing which we expect to pass upon merge of  [partiql-tests/#18](https://github.com/partiql/partiql-tests/pull/18).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-parser/src/parse/partiql.lalrpop [hunk 1]
```diff
@@ -314,15 +319,20 @@ TableQualifiedJoin: ast::FromClauseAst = {
     // Note the `TableReference` on the lhs and the `TableNonJoin` on the rhs of the `JOIN`.
```
Comment (jpschorr) [inline partiql-parser/src/parse/partiql.lalrpop:319]: Need to update this comment

Comment (am357) [reply partiql-parser/src/parse/partiql.lalrpop]: Done.

File: partiql-parser/src/parse/partiql.lalrpop [hunk 2]
```diff
@@ -1093,6 +1104,12 @@ LiteralNumber: ast::Lit = {
 LiteralIon: ast::Lit = {
     <ion:"Ion"> => ast::Lit::IonStringLit(ion.to_owned()),
 }
+#[inline]
+LiteralDateTime: ast::Lit = {
+    <ion:"DATE"> <s:"String"> => ast::Lit::DateTimeLit(ast::DateTimeLit::DateLit(s.to_owned())),
```
Comment (jpschorr) [inline partiql-parser/src/parse/partiql.lalrpop:1109]: I think the `<ion:` can be removed from all of these.

**Review by jpschorr — APPROVED** [review]

File: partiql-parser/src/parse/partiql.lalrpop (hunk 2 above)
Comment (am357) [reply partiql-parser/src/parse/partiql.lalrpop]: I agree, done.

**Pull request merged**